docs/block-client-apis/checkout/checkout-api.md

```diff
@@ -0,0 +1,112 @@
+- `isAfterProcessing`: When the checkout status is `AFTER_PROCESSING` this flag is true. Checkout will have this status after the the block receives the response from the server side processing request.
+- `isComplete`: When the checkout status is `COMPLETE` this flag is true. Checkout will have this status after all observers on the events emitted during the `AFTER_PROCESSING` status are completed successfully. When checkout is at this status, the shopper's browser will be redirected to the value of `redirectUrl` at that point (usually the `order-received` route).
+- `isCalculating`: This is true when the total is being re-calculated for the order. There are numerous things that might trigger a recalculation of the total: coupons being added or removed, shipping rates updated, shipping rate selected etc. This flag consolidates all activity that might be occurring (including requests to the server that potentially affect calculation of totals). So instead of having to check each of those individual states you can reliably just check if this boolean is true (calculating) or false (not calculating).
+- `hasError`: This is true when anything in the checkout has created an error condition state. This might be validation errors, request errors, coupon application errors, payment processing errors etc.
+- `redirectUrl`: The current set url that the checkout will redirect to when it is complete.
+- `onCheckoutAfterProcessingWithSuccess`: Used to register observers that will be invoked after checkout has been processed by the server successfully.
+- `onCheckoutAfterProcessingWithError`: Used to register observers that will be invoked after checkout has been processed by the server and there was an error.
+- `onCheckoutBeforeProcessing`: Used to register observers that will be invoked after the checkout has been submitted but before the processing request is sent to the server.
+- `dispatchActions`: This is an object with various functions for dispatching status in the checkout. It is not exposed to extensions but is for internal use only.
+- `orderId`: The order id for the order attached to the current checkout.
+- `isCart`: This is true if the cart is being viewed. Note: usage of `CheckoutProvider` will automatically set this to false. There is also a `CartProvider` that wraps children in the `ShippingDataProvider` and exposes the same api as checkout context. The `CartProvider` automatically sets `isCart` to true. This allows components that implement `useCheckoutContext` to use the same api in either the cart or checkout context but still have specific behaviour to whether `isCart` is true or not.
+- `hasOrder`: This is true when orderId is truthy.
+- `customerId`: The id of the customer if it's been created yet.
+
+## Hooks
+
+I'm not going to go into detail for all the hooks as that is fairly straightforward from existing implementations. However I do want to highlight an important extension interface hook, `usePaymentMethodInterface`.
```

Comment:
Docs shouldn't be in 1st person.

Reply:
Oops ya this is a copy/paste artifact from the pull description I copied from. I'll fix that in other places too.